### PR TITLE
Add Red Hat's DNS service capability

### PR DESCRIPTION
**Pull request by jtomasek**
assisted-dns

### CODE REVIEW

File: src/components/clusterConfiguration/ClusterConfiguration.tsx [hunk 1]
```diff
@@ -1,332 +1,38 @@
...
+  const [domains, setDomains] = React.useState<ManagedDomain[] | undefined>();
+
+  React.useEffect(() => {
+    const fetchManagedDomains = async () => {
+      try {
+        const { data } = await getManagedDomains();
+        setDomains(data);
+      } catch (e) {
+        setDomains([]);
+        handleApiError(e);
```
Comment (rawagner) [inline src/components/clusterConfiguration/ClusterConfiguration.tsx:23]: Shouldnt we also notify user that we could not fetch the domains via alert `dispatchAlertsAction` ?

Comment (jtomasek) [reply src/components/clusterConfiguration/ClusterConfiguration.tsx:23]: The list of domains is not something that user can influence, so I opted to fail silently (by logging the error in the console) and fall back to empty list of domains. If the list of domains is empty user won't see the option to select to use the RH provided domains.

File: src/components/clusterConfiguration/utils.ts [hunk 2]
```diff
@@ -63,4 +66,7 @@ export const getInitialValues = (cluster: Cluster): ClusterConfigurationValues =
   sshPublicKey: cluster.sshPublicKey || '',
   isPullSecretEdit: !cluster.pullSecretSet, // toggles edit mode and drives validation
   hostSubnet: findMatchingSubnet(cluster.ingressVip, cluster.apiVip, getHostSubnets(cluster)),
+  useRedHatDnsService:
+    !!cluster.baseDnsDomain &&
+    managedDomains.map((d: ManagedDomain) => d.domain).includes(cluster.baseDnsDomain),
```
[the remark this answers is not in the document]
Comment (jtomasek) [reply src/components/clusterConfiguration/utils.ts:71]: Done

File: src/components/clusterConfiguration/NetworkConfiguration.tsx [hunk 3]
```diff
@@ -22,11 +29,58 @@ const NetworkConfiguration: React.FC<NetworkConfigurationProps> = ({ hostSubnets
...
+          helperText="A base DNS domain will be provided by Red Hat's DNS service. Because the cluster's DNS can't be changed later, this should only be used for temporary, non-production clusters."
+          onChange={toggleRedHatDnsService}
+        />
+      )}
+      {values.useRedHatDnsService ? (
+        <SelectField
+          label="Base DNS Domain"
+          name="baseDnsDomain"
+          helperText={baseDnsHelperText}
+          options={managedDomains.map((d: ManagedDomain) => ({
```
[the remark this answers is not in the document]
Comment (jtomasek) [reply src/components/clusterConfiguration/NetworkConfiguration.tsx:68]: Done

File: src/components/clusterConfiguration/NetworkConfiguration.tsx [hunk 4]
```diff
@@ -22,11 +29,58 @@ const NetworkConfiguration: React.FC<NetworkConfigurationProps> = ({ hostSubnets
...
+    <>
+      The base domain of the cluster. All DNS records must be sub-domains of this base and include
+      the cluster name. This cannot be changed later. The full cluster address will be:{' '}
+      <strong>
+        {clusterName || '[Cluster Name]'}.{baseDnsDomain || '[example.com]'}
+      </strong>
+    </>
+  );
+
+  const toggleRedHatDnsService = (checked: boolean) => {
```
[the remark this answers is not in the document]
Comment (jtomasek) [reply src/components/clusterConfiguration/NetworkConfiguration.tsx:42]: Done

File: src/components/clusterConfiguration/utils.ts [hunk 5]
```diff
@@ -63,4 +66,6 @@ export const getInitialValues = (cluster: Cluster): ClusterConfigurationValues =
   sshPublicKey: cluster.sshPublicKey || '',
   isPullSecretEdit: !cluster.pullSecretSet, // toggles edit mode and drives validation
   hostSubnet: findMatchingSubnet(cluster.ingressVip, cluster.apiVip, getHostSubnets(cluster)),
+  useRedHatDnsService:
+    !!cluster.baseDnsDomain && managedDomains.map((d) => d.domain).includes(cluster.baseDnsDomain),
```
Comment (rawagner) [inline src/components/clusterConfiguration/utils.ts:70]: What happens if I 
1. create a cluster
2. set my custom dns domain
3. save and close the form
4. open the form again
 ? 
Seems like `useRedHatDnsService` will be true, which is not correct ?

File: src/components/clusterConfiguration/ClusterConfiguration.tsx (hunk 1 above)
Comment (rawagner) [reply src/components/clusterConfiguration/ClusterConfiguration.tsx:23]: Its true that user cannot influence it, but he still may be left wondering what happened to managed domains ? I still prefer transparency over failing silently. I think any request that fails and shouldn't  (meaning its an actual error) should be visualized to user. But thats just my preference, if you think the current solution is better then lets go with it.

File: src/components/clusterConfiguration/utils.ts (hunk 5 above)
Comment (rawagner) [reply src/components/clusterConfiguration/utils.ts:70]: ok, somehow I missed the includes :)

File: src/components/clusterConfiguration/ClusterConfiguration.tsx (hunk 1 above)
Comment (jtomasek) [reply src/components/clusterConfiguration/ClusterConfiguration.tsx]: OK, I've enhanced the error handling to include errors for fetching the domains.

**Pull request merged**